src/software/ai/navigator/path_planner/hrvo/hrvo_agent.h

```diff
@@ -220,20 +231,38 @@ class HRVOAgent : public Agent
     std::vector<ObstaclePtr> static_obstacles;
     std::optional<ObstaclePtr> ball_obstacle;
 
-    // pointers to the closest agents by euclidean distance to this agents position
+    // pointers to the closest agents by Euclidean distance to this agents position
     std::vector<std::shared_ptr<Agent>> neighbours;
 
-    // scale used to normalise preferred velocity so that agent has ability to speed up as
-    // a way to avoid obstacles
-    static constexpr double PREF_SPEED_SCALE = 0.85;
+    Point prev_dynamic_kp_destination;
+    double kp;
+
+    // The maximum amount that the destination can be moved (in meters) before
+    // we recalculate the dynamic kp value
+    static constexpr double MAX_DESTINATION_CHANGE_THRESHOLD = 0.1;
+
+    // The maximum amount which the PID velocity can be greater than
+    // the actual robot velocity. Used to prevent swinging [m/s]
+    static constexpr double LINEAR_VELOCITY_MAX_PID_OFFSET = 0.3;
+
+    // Proportional constant used for calculating the desired angular velocity
+    static constexpr double ANGULAR_VELOCITY_KP = 4.5;
+
+    // Compensation constant added to stop the robots from swinging when turning and
+    // moving in a linear line. This is defined as a multiplier of the angular velocity
+    // for how much the linear velocity vector should be rotated by in the opposite
+    // direction of the angular velocity.
+    static constexpr double ANGULAR_VELOCITY_COMPENSATION_MULTIPLIER = 1.5;
 
-    // a multiplier to account for different accelerations when speeding up vs slowing
-    // down
-    static constexpr double DECEL_DIST_MULTIPLIER = 1.2;
+    // The obstacle factory robot_obstacle_inflation_factor used by HRVO
+    // This is lower than the default value in the config used by path planner since HRVO
+    // automatically takes into account the robot radius when avoiding obstacles.
+    static constexpr double HRVO_STATIC_OBSTACLE_INFLATION_FACTOR = 0.5;
```

Comment:
could this be a dynamic param?

Reply:
Umm potentially. I haven't really thought about it. The main thing would be that the path planner uses a constant value greater than 0.5 (it uses something around 2), so even if we make this 0, the paths will not take the robot any closer (for static obstacles at least)